addons/website/models/ir_ui_view.py

```diff
@@ -522,7 +522,19 @@ def save(self, value, xpath=None):
                 ('website_id', '=', current_website.id)
             ], limit=1)
             if website_specific_view:
-                self = website_specific_view
+                if (
+                    website_specific_view.page_ids
+                    and website_specific_view.page_ids[0].url != self.page_ids[0].url
+                ):
+                    # The case here is when a generic page is edited after its
+                    # specific page has a different URL. In this case, the
+                    # generic page can still be accessed since the specific one
+                    # does not shadow it anymore. In such a case, we need the
+                    # write to be done on the edited generic page and not target
+                    # the specific one.
+                    self = self.with_context(no_cow=True)
```

Comment:
Not sure I like this, IMO once you change the URL of the generic page it should not be available for your current website anymore (the routing should see that you made a specific page with a different URL). So this case here should not exist. IMO the flow may look more buggy than before:
- Go to website 1 /contactus page
- Enter edit mode, change the URL used as email_to for the form, save
- Open page properties, change the URL
- Go to /contactus again, you land on the generic page... but you have no way of knowing it, it's just a different email. Enter edit mode, add a banner at the top... now you edited all your ten websites but not the one you actually wanted clicking on "Contact us" from the page manager, not even looking at the URL :shrug: 

I guess this code could exist as a fallback for a case which is not supposed to exist... but the main problem IMO is being able to land on the generic page whose URL was changed. And I guess that if we do want to implement that, we cannot indeed change the keys (even in master). In fact why is it not already working if we base the "is there a specific view?" on keys?

Who knows, maybe in master, we should not allow the generic page feature entirely. As you said, probably bad as SEO always :shrug: So we could have something like "when creating a website, a page is created for contact us and homepage but they are specific without generic automatically" :shrug: Although, I am sure the PS tech could have used that feature, bad for SEO or not? :shrug:

Reply:
> Although, I am sure the PS tech could have used that feature, bad for SEO or not? 🤷

Yup, it's probably something we need to keep, it's probably very practical in some cases where you don't care about SEO or don't even index the page.

> IMO the flow may look more buggy than before:

I don't think it's worst than keep trying to save changes that are actually looking like they are never saved and you keep losing your changes (while it is just writing on another page behind the scene), which is what the 2-3 people complained when they reported the issue, but yeah, it's probably in both cases at the very high of the "terrible UX" scale :grimacing: 

Even if I think you mostly do visual diff (as explained in my step to reproduce and reported by people) than hidden diff (your case about email option), so it really looks "wtf is going on".

I'll try to figure another way, to give the editor a warning upfront like a banner or something on the generic page when we detect a specific counterpart.
Or maybe try to alter the routing to not let you access the page at all once it's "COW'd" and url changed.
Even if it seems very weak and fragile to consider such pages as "Generic and specific to each other" just because they got the same key..
For the user, they changed their url and their content, there is nothing left in common between the pages.

It's why I made the fix that way, for me it seems quite correct to say "you have modified the page URL, and most of the time it's content too, those are different pages now".

So about what you say here:

> Go to /contactus again, you land on the generic page... but **you have no way of knowing it**, it's just a different email.

Imo there is nothing to know, you made those pages independent and not related anymore. You most of the time will see a visual diff and should notice it.

But I got your opinion.. Not an easy one :grimacing: 
I think the banner is the way to go: you can access the page, but the system is letting you know that this is linked to another specific page. We will need to find a good wording.

Or even simpler, when entering edit mode on such a generic page shadowed by a specific page (again, specific for odoo technical code, but not for lambda users, those have nothing in common anymore), we show an alert to the user (could be blocking).

Let's discuss this tomorrow IRL once you've read this to see what we decide?